avocado/core/runner.py

```diff
@@ -377,6 +376,20 @@ def sigtstp_handler(signum, frame):     # pylint: disable=W0613
             test_state = test_status.abort(proc, time_started, cycle_timeout,
                                            first, step)
 
+        # Try to log the timeout reason to test's results and update test_state
+        if abort_reason:
+            TEST_LOG.error(abort_reason)
+            test_log = test_state.get("logfile")
+            if test_log:
+                open(test_log, "a").write("\nRUNNER: " + abort_reason + "\n")
```

Comment:
I really wish there was a cleaner way to write this message to the test logger without crossing this boundary... but doesn't look like it's possible. One of those things that are easier said than done.  Opening in `a` mode is a nice safeguard.  All in all, OK!